test/utils/helpFn.ts

```diff
@@ -47,40 +55,38 @@ export const getContext = async (polkadotApi: ApiPromise) => {
   * @param extrinsicHex 
   * @returns promise
   */
- export const sendExtrinsic = async (polkadotApi: ApiPromise, extrinsicHex: HexString) : Promise<{extrinsicHash: string, blockHash: string, events: any[]}> => {
-   return new Promise(async (resolve, reject) => {
-     try {
-       const txHash = await sendExtrinsicToChain(polkadotApi, extrinsicHex, ({ status, events, dispatchError }) => {
-         if (status?.isFinalized) {
-           if (!_.isNil(dispatchError)) {
-             if (dispatchError.isModule) {
-                 const metaError = polkadotApi.registry.findMetaError(dispatchError.asModule);
-                 const { name, section } = metaError;
-                 reject(new Error(`${section}.${name}`));
-                 return;
-             } else {
-                 reject(new Error(dispatchError.toString()));
-                 return;
-             }
-           }
- 
-           const event = _.find(events, ({ event }) => polkadotApi.events.system.ExtrinsicSuccess.is(event));
-           // We pass the event array out so the caller can access and fetch relevant data in extrinsic such as the task id. Recall that task id is emit from the TaskScheduled { who, taskId } event
-           // if the event is empty that mean the extrinsic failed the test resolve to error
-           if (event) {
-             resolve({ extrinsicHash: txHash, blockHash: status?.asFinalized?.toString(), events });
-           } else {
-             reject(new Error('The event.ExtrinsicSuccess is not found'));
-           }
-         }
-       });
-     } catch (error) {
-       reject(error);
-     }
-   });
+export const sendExtrinsic = async (polkadotApi: ApiPromise, extrinsicHex: HexString) : Promise<{extrinsicHash: string, blockHash: string, events: any[]}> => {
+  return new Promise((resolve, reject) => {
+    const extrinsic = polkadotApi.tx(extrinsicHex);
+    sendExtrinsicToChain(polkadotApi, extrinsicHex, ({ status, events, dispatchError }) => {
+      if (status?.isFinalized) {
+        if (!_.isNil(dispatchError)) {
```

Comment:
_.isNil(dispatchError) is almost different than !dispatchError, which doesn’t distinguish undefined, null and 0.

**Please, just replace all `!variable` and `!_.isNil` with `_.isUndefined` in this PR !!**

Reply:
No, this is not fixed. I meant to **replace all `!_.isNil` with `!_.isUndefined`**, unless you are very clear about a specific error case.